src/content_cao.cpp

```diff
@@ -1786,6 +1831,14 @@ void GenericCAO::processMessage(const std::string &data)
 	}
 }
 
+void GenericCAO::knockback(const v3f &direction, f32 time_knockback)
+{
+	m_knockback_timer = time_knockback;
```

Comment:
No need for a timer, just throw the player in a ballistic trajectory until they hit something.